includes/admin/class-wc-subscriptions-admin.php

```diff
@@ -2016,10 +2009,9 @@ public static function insert_setting_after( &$settings, $insert_after_setting_i
 	 * @param array $settings The list of settings
 	 */
 	public static function add_guest_checkout_setting_note( $settings ) {
-		$is_wc_pre_3_4_0 = wcs_is_woocommerce_pre( '3.4.0' );
-		$current_filter  = current_filter();
+		$current_filter = current_filter();
 
-		if ( ( $is_wc_pre_3_4_0 && 'woocommerce_payment_gateways_settings' !== $current_filter ) || ( ! $is_wc_pre_3_4_0 && 'woocommerce_account_settings' !== $current_filter ) ) {
+		if ( 'woocommerce_account_settings' !== $current_filter ) {
```

Comment:
(2/2) And call the function directly:

```suggestion
		if ( 'woocommerce_account_settings' !== current_filter() ) {
```

Imho, this is just as readable, and shaves off some extra lines of code (but again, not strictly needed, either).